soc/nordic/common/dmm.h

```diff
@@ -0,0 +1,96 @@
+/*
+ * Copyright (c) 2024 Nordic Semiconductor ASA
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/**
+ * @file
+ * nRF SoC specific public APIs for Device Memory Management (dmm) subsystem
+ */
+
+#ifndef SOC_NORDIC_COMMON_DMM_H_
+#define SOC_NORDIC_COMMON_DMM_H_
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/** @cond INTERNAL_HIDDEN */
+
+/**
+ * @brief Prepare a DMA output buffer for the specified device
+ *
+ * Allocate an output buffer in memory region that given device can perform DMA transfers from.
+ * Copy @p user_buffer contents into it.
+ * Writeback data cache lines associated with output buffer, if needed.
+ *
+ * @note Depending on provided user buffer parameters and SoC architecture,
+ *       dynamic allocation and cache operations might be skipped.
+ *
+ * @warning It is prohibited to read or write @p user_buffer or @p buffer_out contents
+ *          until @ref dmm_dma_buffer_out_release() is called
+ *
+ * @param dev Device to prepare the buffer for.
+ * @param user_buffer Buffer containing data to be processed by the given device
+ * @param user_out Pointer to a pointer to be filled by prepared DMA buffer address.
+ *
+ * @retval 0 If succeeded.
+ * @retval -ENOMEM If output buffer could not be allocated.
+ * @retval -errno Negative errno for other failures.
+ */
+int dmm_dma_buffer_out_prepare(const struct device *dev, void const * user_buffer, void ** buffer_out);
```

Comment:
it would be good  to discuss whether `{in,out}` function variants should be merged into one with additional parameter like `direction={IN/OUT}`

Reply:
I don't think so. These functions are used in different contexts, so it is cleaner to keep them with different symbols. It's easier to follow the code this way.